src/rosdiscover/project/models.py

```diff
@@ -60,30 +61,37 @@ def close(self) -> None:
 
     def _recover(self, package: str, node: str) -> t.Optional[NodeModel]:
         # have we already recovered this model?
+        print(f"Recovering {package}/{node}")
         if self._recovered_models.contains(self.config, package, node):
             return self._recovered_models.fetch(self.config, package, node)
 
         # is this node model irrecoverable?
         if (package, node) not in self.config.node_sources:
-            return None
-
-        # TODO we need to know the sources for this node
-        # - eventually, we want this to come from CMakeLists
-        # - for now, we can manually specify these as part of the configuration
-        node_info = self.config.node_sources[(package, node)]
-
-        # use the recovery tool to recover the model before saving it to the database
-        model = self._recovery_tool.recover(
-            package,
-            node,
-            node_info.entrypoint,
-            node_info.sources,
-            node_info.restrict_to_paths
-        )
+            try:
+                logger.info(f"Attempting to recover {package}/{node} from CMakeLists.txt")
+                model = self._recovery_tool.recover_using_cmakelists(package, node)
+            except ValueError as e:
```

Comment:
We may also want to catch `RuntimeError` here, too. (We use that in cases where the input is fine but the C++ static recovery failed.)

Reply:
Done